src/main/java/org/radarcns/redcap/integration/Integrator.java

```diff
@@ -58,9 +57,18 @@ public boolean handleDataEntryTrigger(){
         String integrationFrom = redCapInfo.getIntegrationForm();
         URL url = redCapInfo.getUrl();
         List<String> attributeKeys = redCapInfo.getAttributeFieldNames();
+
         Map<String, String> attributes = redCapIntegrator.pullRecordAttributes(attributeKeys, recordId);
-        Subject subject = mpIntegrator.performSubjectUpdateOnMp(url, projectId, recordId, attributes);
-        return redCapIntegrator.updateRedCapIntegrationForm(subject, recordId, enrolmentEvent, integrationFrom);
+        String redcapSubjectId = redCapIntegrator.pullRecordSubjectId(recordId);
+        Subject subject = mpIntegrator.performSubjectUpdateOnMp(url, projectId, recordId, attributes, redcapSubjectId);
+        Subject.SubjectOperationStatus result = subject.getOprationStatus();
+        if(result.equals(Subject.SubjectOperationStatus.CREATED) ){
+            return redCapIntegrator.updateRedCapIntegrationForm(subject, recordId, enrolmentEvent, integrationFrom);
+        }
+        else {
+            return (result.equals(Subject.SubjectOperationStatus.UPDATED) || result.equals(Subject.SubjectOperationStatus.OTHER))
+                    && !result.equals(Subject.SubjectOperationStatus.FAILED);
```

Comment:
The POST request handler (in EntryPoint class) response depends on the result here. Should I remove the error response there if it returns false and only return an error response on exceptions?

Reply:
> Should I remove the error response there if it returns false and only return an error response on exceptions?
No, that should be fine. 

If you get a `Failed` status here, you can just throw an IllegalStateException and if anything else, it should return false (since there was no error but we still did not update the redcap form)

Reply:
Okay. Made the changes.